test/versioned/restify/restify-post-7/async-handlers.tap.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+
+const helper = require('../../../lib/agent_helper')
+
+const wait = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+const randomNumber = () => Math.floor(Math.random() * 1000)
+
+const getRandomNumber = async () => {
+  const number = randomNumber()
+  await wait(number / 10)
+  return number
+}
+
+tap.test('Restify instrumentation', (t) => {
+  t.autoend()
+
+  let agent = null
+  let restify = null
+  let server = null
+  const numbers = []
+  let i = 0
+
+  t.before(() => {
+    agent = helper.instrumentMockedAgent()
+    restify = require('restify')
+    server = restify.createServer()
+
+    const useSyncMiddleware = (req, res, next) => {
+      numbers[i] = randomNumber()
+      i++
+      const txn = agent.getTransaction()
+      t.ok(txn, `sync middleware should be in transaction context (${req.method} ${req.path()})`)
```

Comment:
maybe instead of asserting in these tests, incrementing when it is in a transaction in every middleware and then in the tests assert the expected count?